Assets/BossRoom/Prefabs/UI/SettingsPanelCanvas.prefab

```diff
@@ -2571,136 +2665,168 @@ PrefabInstance:
   m_Modification:
     m_TransformParent: {fileID: 6417648243838381818}
     m_Modifications:
-    - target: {fileID: 7721113380232533813, guid: addf3685293c35d48a673397c748f116, type: 3}
+    - target: {fileID: 7721113380232533813, guid: addf3685293c35d48a673397c748f116,
+        type: 3}
       propertyPath: m_text
       value: Start
       objectReference: {fileID: 0}
-    - target: {fileID: 7721113380319664578, guid: addf3685293c35d48a673397c748f116, type: 3}
+    - target: {fileID: 7721113380319664578, guid: addf3685293c35d48a673397c748f116,
+        type: 3}
       propertyPath: m_Name
       value: Settings Button
       objectReference: {fileID: 0}
-    - target: {fileID: 7721113380319664588, guid: addf3685293c35d48a673397c748f116, type: 3}
+    - target: {fileID: 7721113380319664588, guid: addf3685293c35d48a673397c748f116,
+        type: 3}
       propertyPath: m_OnClick.m_PersistentCalls.m_Calls.Array.size
       value: 1
       objectReference: {fileID: 0}
-    - target: {fileID: 7721113380319664588, guid: addf3685293c35d48a673397c748f116, type: 3}
+    - target: {fileID: 7721113380319664588, guid: addf3685293c35d48a673397c748f116,
+        type: 3}
       propertyPath: m_OnClick.m_PersistentCalls.m_Calls.Array.data[0].m_Mode
       value: 1
       objectReference: {fileID: 0}
-    - target: {fileID: 7721113380319664588, guid: addf3685293c35d48a673397c748f116, type: 3}
+    - target: {fileID: 7721113380319664588, guid: addf3685293c35d48a673397c748f116,
+        type: 3}
       propertyPath: m_OnClick.m_PersistentCalls.m_Calls.Array.data[0].m_Target
       value: 
       objectReference: {fileID: 811129276045199873}
-    - target: {fileID: 7721113380319664588, guid: addf3685293c35d48a673397c748f116, type: 3}
+    - target: {fileID: 7721113380319664588, guid: addf3685293c35d48a673397c748f116,
+        type: 3}
       propertyPath: m_OnClick.m_PersistentCalls.m_Calls.Array.data[0].m_CallState
       value: 2
       objectReference: {fileID: 0}
-    - target: {fileID: 7721113380319664588, guid: addf3685293c35d48a673397c748f116, type: 3}
+    - target: {fileID: 7721113380319664588, guid: addf3685293c35d48a673397c748f116,
+        type: 3}
       propertyPath: m_OnClick.m_PersistentCalls.m_Calls.Array.data[0].m_MethodName
       value: SetQualitySettings
       objectReference: {fileID: 0}
-    - target: {fileID: 7721113380319664588, guid: addf3685293c35d48a673397c748f116, type: 3}
+    - target: {fileID: 7721113380319664588, guid: addf3685293c35d48a673397c748f116,
+        type: 3}
       propertyPath: m_OnClick.m_PersistentCalls.m_Calls.Array.data[0].m_TargetAssemblyTypeName
       value: QualityButton, Assembly-CSharp
```

Comment:
I believe the text component here got dereferenced. There's a null ref when you try to change quality settings.